cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/schedule/AbstractSchedulerService.java

```diff
@@ -168,11 +168,18 @@ public void schedule(Id.Program programId, SchedulableProgramType programType, S
 
     scheduler.schedule(programId, programType, schedule, properties);
     if (isLazyStart()) {
-      try {
-        scheduler.suspendSchedule(programId, programType, schedule.getName());
-      } catch (NotFoundException e) {
-        // Should not happen - we just created it. Could have been deleted just in between
-        LOG.info("Schedule could not be suspended - it did not exist: {}", schedule.getName());
+      // TODO: CDAP-2281 figure out a better way to handle schedules in unit tests
+      String ignoreLazy = properties.get(Constants.Scheduler.IGNORE_LAZY_START);
+      boolean scheduleIgnoresLazy = ignoreLazy != null && Boolean.valueOf(ignoreLazy);
+      if (scheduleIgnoresLazy) {
+        lazyStart(scheduler);
```

Comment:
where was lazyStart being called before this change? I couldn't find it in the code removed below.

nit: Also this check and method name is confusing, if "schedule ignores lazy", then "lazy start the schedule".